.github/workflows/azure-e2e-test.yml

```diff
@@ -0,0 +1,177 @@
+    steps:
+    - uses: actions/checkout@v2
+
+    - uses: ./.github/actions/login-to-azure
+      with:
+        client-id: ${{ secrets.AZURE_CLIENT_ID }}
+        client-secret: ${{ secrets.AZURE_CLIENT_SECRET }}
+        subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+        tenant-id: ${{ secrets.AZURE_TENANT_ID }}
+
+    - name: Docker login
+      run: |
+        TOKEN=$(az acr login --name ${{ secrets.ACR_URL }} --expose-token --output tsv --query accessToken)
+        docker login ${{ secrets.ACR_URL }} --username 00000000-0000-0000-0000-000000000000 --password "$TOKEN"
+
+    - name: Build and push CAA docker image
+      env:
+        registry: ${{ secrets.ACR_URL }}
+        CLOUD_PROVIDER: azure
+      run: |
+        docker buildx create --name multiarch --driver docker-container --use
+        make image
+
+  build-podvm-image:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+
+    - uses: ./.github/actions/login-to-azure
+      with:
+        client-id: ${{ secrets.AZURE_CLIENT_ID }}
+        client-secret: ${{ secrets.AZURE_CLIENT_SECRET }}
+        subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+        tenant-id: ${{ secrets.AZURE_TENANT_ID }}
+
+    - uses: ./.github/actions/setup-azure-build-env
+
+    - name: Clone kata repository
+      run: |
+        pushd ../
+        git clone https://github.com/kata-containers/kata-containers -b CCv0
+        popd  
+
+    - name: Create podvm image
+      id: create-podvm-image
+      env:
+        PKR_VAR_subscription_id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+        PKR_VAR_client_id: ${{ secrets.AZURE_CLIENT_ID }}
+        PKR_VAR_client_secret: ${{ secrets.AZURE_CLIENT_SECRET }}
+        PKR_VAR_tenant_id: ${{ secrets.AZURE_TENANT_ID }}
+        PKR_VAR_resource_group: ${{ secrets.AZURE_RESOURCE_GROUP }}
+        PKR_VAR_location: ${{ secrets.AZURE_REGION }}
+        PKR_VAR_az_image_name: ${{ env.PODVM_IMAGE_NAME }}
+        PKR_VAR_vm_size: ${{ env.VM_SIZE }}
+        PKR_VAR_ssh_username: ${{ env.SSH_USERNAME }}
+        PKR_VAR_az_gallery_name: ${{ env.PODVM_GALLERY_NAME }}
+        PKR_VAR_az_gallery_image_name: ${{ env.PODVM_IMAGE_DEF_NAME }}
+        PKR_VAR_az_gallery_image_version: ${{ env.PODVM_IMAGE_VERSION }}
+        GOPATH: /home/runner/go
+        PODVM_DISTRO: "ubuntu"
+      run: |
+        pushd azure/image
+        make image
+        popd
+        
+  run-e2e-test:
+    runs-on: ubuntu-latest
+      #needs:
+      - build-caa-image
+      - build-podvm-image
+    steps:
+    - uses: actions/checkout@v2
+
+    - uses: ./.github/actions/login-to-azure
+      with:
+        client-id: ${{ secrets.AZURE_CLIENT_ID }}
+        client-secret: ${{ secrets.AZURE_CLIENT_SECRET }}
+        subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+        tenant-id: ${{ secrets.AZURE_TENANT_ID }}
+
+    - name: Create provisioner file
+      env:
+        AZURE_SUBSCRIPTION_ID: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
```

Comment:
I guess there is no reason to put the variables into env's we can just use them in the cat << EOF directly.

Reply:
can we move the envs into the ´cat EOF >>` expression directly to avoid duplication?